src/elements/segment.rs

```diff
@@ -123,8 +172,21 @@ impl Serialize for DataSegment {
 	type Error = Error;
 
 	fn serialize<W: io::Write>(self, writer: &mut W) -> Result<(), Self::Error> {
-		VarUint32::from(self.index).serialize(writer)?;
-		self.offset.serialize(writer)?;
+		if self.passive {
+			VarUint7::from(1).serialize(writer)?;
+		} else if self.index != 0 {
+			VarUint7::from(2).serialize(writer)?;
+			VarUint32::from(self.index).serialize(writer)?;
+		} else {
+			VarUint7::from(0).serialize(writer)?;
+		}
+		match self.offset {
+			Some(offset) => {
+				assert!(!self.passive);
```

Comment:
parity-wasm should never panic on any input
is it a possible panic?

Reply:
Sure thing, updated!